tests/unit/test_py_vs_cpp.py

```diff
@@ -0,0 +1,130 @@
+import inspect
+import re
+from unittest import mock
+
+import pytest
+from antlr4 import InputStream, ParserRuleContext
+from antlr4.Token import CommonToken
+from antlr4.tree.Tree import TerminalNodeImpl
+
+from pytsql.grammar import sa_tsql
+from pytsql.tsql import _split
+
+
+@pytest.fixture
+def seed():
+    return """
+DECLARE @timestamp_started  DATETIME  = CURRENT_TIMESTAMP;
+-- step 0: deduce associated nr's
+DROP TABLE IF EXISTS #temporary;
+CREATE TABLE #temporary (
+    nr      VARCHAR(17)   NOT NULL
+  , short_nr         SMALLINT NOT NULL
+  , min_nr_date     DATE     NOT NULL
+  , iter        TINYINT  NOT NULL -- iterator
+  , PRIMARY KEY (nr, min_date, iter)
+);
+PRINT(CAST(GETDATE() AS VARCHAR) + ': INSERT INTO #temporary')
+INSERT INTO #temporary WITH(TABLOCK)
+SELECT CONCAT('0', nr) nr
+     , short_nr
+     , MIN(nr_date) min_nr_date
+     , ROW_NUMBER() OVER(PARTITION BY nr, MIN(nr_date) ORDER BY short_nr DESC) -- very few instances where this matters
+FROM otherdb.dbo.othertable WITH(NOLOCK)
+WHERE kind = 1
+GROUP BY nr, short_nr;"""
+
+
+def get_rule_labels(context_cls):
+    init_func = context_cls.__init__
+
+    # Detect any context/token labels from the init function's assignments
+    # Conveniently, the type name is provided as a comment
```

Comment:
Do we use the type comments in any way? I'd remove the comment if not.
(I know this is copied from the original validator, but I'm in favor of having the least amount of comments in code)

Reply:
It seems we don't use the type names provided in the comments. I'll remove this comment.
```suggestion
```